x/axelarcork/client/cli/tx.go

```diff
@@ -486,28 +495,33 @@ Where proposal.json contains:
 				return err
 			}
 
-			proposal, err := ParseCommunityPoolSpendProposal(clientCtx.Codec, args[0])
+			proposal := types.AxelarCommunityPoolSpendProposalForCLI{}
+			contents, err := os.ReadFile(args[0])
 			if err != nil {
 				return err
 			}
 
-			if len(proposal.Title) == 0 {
-				return fmt.Errorf("title is empty")
+			if err = clientCtx.Codec.UnmarshalJSON(contents, &proposal); err != nil {
+				return err
 			}
 
-			if len(proposal.Description) == 0 {
-				return fmt.Errorf("description is empty")
+			if proposal.ChainId == 0 {
+				return fmt.Errorf("chain ID must be non-zero")
 			}
 
 			if !common.IsHexAddress(proposal.Recipient) {
-				return fmt.Errorf("recipient is not a valid Ethereum address")
+				return fmt.Errorf("recipient is not a valid EVM address")
 			}
 
 			amount, err := sdk.ParseCoinNormalized(proposal.Amount)
 			if err != nil {
 				return err
 			}
 
+			if !amount.IsPositive() {
+				return fmt.Errorf("amount must be positive")
+			}
+
 			deposit, err := sdk.ParseCoinsNormalized(proposal.Deposit)
 			if err != nil {
 				return err
```

Comment:
In other commands, we run ValidateBasic on `content`, but here we do the checks before created the `content` var. Is this intentional?

Reply:
There was one case where I recall using `ValidateBasic` because it was much simpler than repeating some stuff, but formats of input/output aren't always the same for the CLI fields vs. the message (e.g. everything coming in is a string we have to convert). Checks made in the CLI before it gets submitted avoids having a failed TX, which short circuits the process faster. We don't really want to duplicate all the validation checks, but they're needed in the chain, and adding some here is just better user experience.